integration-tests/src/main/java/org/apache/druid/testing/utils/KinesisAdminClient.java

```diff
@@ -156,15 +160,35 @@ public boolean verfiyPartitionCountUpdated(String streamName, int oldShardCount,
     return actualShardCount == oldShardCount + newShardCount;
   }
 
+  private List<Shard> listShards(String streamName)
```

Comment:
1. why have this in a separate method instead of inlining it in the only caller?
2. using Arraylist here whereas the main method uses Set - can there be duplicates? if not, then we can only maintain a counter

Reply:
This method was intended for a change which will now be made as part of another ticket